### ISSUE TITLE
FTB Chunk loader breaking other mods

### ISSUE THREAD

**Issue by bound4doom** [LatvianModder/FTBUtilities#360]
Ok, so I have been having lots of issues, it all comes back around to chunk loading. 

I am going to try to be as descriptive as possible but link to other bug issues in other mods as well as they are all trying to fix them but I think the main issue is with the FTB Chunk loader as several other mods are having the same issues.

Basically if a mod has a system that crosses chunk borders. When all players log off, and when they log back on the systems that cross the chunk breaks. People using other chunk loaders have found the other chunk loaders fix this issue.

First let me tell you about the set up so you can try to reproduce. 
Host is www.beastnode.com
I am running the Direwolf 20 pack, 1.3.2, 
We have about 6-7 players that log on occasionally. However it is not uncommon to be the only one on as the others are beta testing some other games right now. So there may be long stretches in time no on is on the server. However, the issue occurs simply if your the only person on the server and you log off and right back on or if the client crashes, by the time you get back on, well you have the problem again.

Example 1: Ender IO Conduits, if you have conduits that cross chunk borders they break.  Not all of them, just some, however the ones that break are always repeatable. 

See this bug report on their repository. It is always with FTB Chunk loader and crossing chunk borders.
https://github.com/SleepyTrousers/EnderIO/issues/3943 

Ok this is my own experience with it. This is a simple one it was repeatable for a long time. When I say it was repeatable more on that later which leads me to go with the chunk loader.

So using an Actually Addition empowerer, to empower canola seeds. Every single time I logged on I would have to break the item pipes and put them back in, only one, but that would fix it. Also it was not both ways just one way.

![image](https://cloud.githubusercontent.com/assets/25747811/23030459/cb807422-f43b-11e6-94ba-a16edfaabdac.png)

So you can see from this setup, green pipes are item pipes, red power. I always have power. I have this set up to throw the seeds needed for this operation into the chest and then place the then empowered seeds back into the chest. When I did this, after signing in, the only one that would get populated would be number 3 on the image. I could go aver and manually place the seeds, they would empoer and get removed fine, and then instantly would place number 3 back in, but none of the others would work.

So after messing around and going to read bug reports on Ender IO. This is where I learned it was a cross border thing. 

Which you can see where when I show borders. The chest and empower number 3 is on one side of the border and the others are on the other side. All the chunks shown are claimed by the way. This was completely repeatable. Every single time.
![image](https://cloud.githubusercontent.com/assets/25747811/23030710/8b5e852c-f43c-11e6-87a0-2f5395b426ec.png)

Ok so in an effort to fix it.  I tried simply unclaiming everything, then claiming it again then chunkloading it again. I logged off and logged back on. Everything worked. 

However now my refined storage system was offline. Odd, so I had to break all the refined storage blocks and place them back in. It all came right back up. I tried this 3 days. Now logging in every day, my empowerer is working just fine but my refined storage is offline.

Which led me to searching on this and well Refined storage has a long bug report on this they are trying to figure out on.

Example 2: Which you guessed it, refined storage is having trouble when it crosses chunk borders. After everyone logs out. 
https://github.com/raoulvdberge/refinedstorage/issues/370

This was again repeatable for me every single day, after I undid and redid all my chunks. So I tried again I undid all my chunks and then reclaimed them and chunkloaded them again. Then logged off and logged on later. Now the pipes to my farm are broken but my refined storage is online. 

So this leads me to believe there is something going on with the chunkloader causing these other mods to break. In the bug reports people have fixed them by using other chunk loaders. However things like chicken chunks and so on are not compatible with direwolf 20. 

Do chunks need to be loaded in a certain order? Does this matter or would it matter? Anything I can try or give you more information?

**Comment by LatvianModder** [comment]
Wow this is so well detailed.. Ok, let me see if I can reproduce the issues myself, and then I can tell more. 
The chunk loading order shouldnt matter, otherwise these pipes would break on a simple world loading, but I could be something similar

**Comment by eobie** [comment]
I can confirm this bug. ExtraUtils fixes the problem.

**Comment by Marcel40625** [comment]
"ExtraUtils fixes the problem." What about ExtraUtils fixes it ? or do you replace EnderIO conduits?

**Comment by eobie** [comment]
I meant the chunkloading was fixed with the extrautil chunkloader. We didn't see the same issue.

**Comment by bound4doom** [comment]
Yeah I started using the Extra Utils Chunk Loading Ward. Chunk loading in the FTB Chunk loader seem to be intermittent. Some times it works sometimes it doesn't. I have not had this issue since using the extra utils Chunk loading ward. The FTB chunkloader is definitely I think one of the root causes to these problems. I took this screen shot the other night. I am trying to gather more evidence of the FTB chunk loader just not working.

The Screen shot below shows my Ender lily field, With Chunk Borders turned on. Ender Lilies planted in in End. Also under Greenhouse glass. They grow very fast. Before I went to bed I picked them all and signed off. This is how they were when I signed on the following day in the late evening. The ones in left hand of the border chunk is under the Extra utils chunk loading ward.  The ones on the right is the FTB Chunk loader.

![2017-02-19_16 01 13_zpsruyro7pm](https://cloud.githubusercontent.com/assets/25747811/23871133/8d01ee56-07ff-11e7-95bd-8069b7abd0af.png)

**Comment by eobie** [comment]
Same thing. Picked ender lillies before I went to bed. Log in come morning time. Still little saplings. Then in an hour they are almost done. Planted in end stone as well.

**Comment by LatvianModder** [comment]
Actually, from what I have tested, seems like chunkloading works, until server restarts. So if you log in once after server has started, log out, and the return after X amount of time, it should have stayed loaded for that time. Can anyone else test this?

**Comment by bound4doom** [comment]
Will test as soon as I can, it will probably be Wednesday with a report on Thursday, if you can wait till weekend I can give you more results as me and the admin will just bounce it a few times. however I can agree and disagree with your findings. 

We do restart server every night. I also do notice quite often things are working if my other friends on the server have been on playing before me and are on when I sign on. They also have different chunks in different parts of the world. However, if they have been on but signed off before I get on, with no restart in between, I find my crops and stuff have all grown, so like them being on allowed chunk loading to work. However by the time I get on and get in. All the pipes are broken and needs reset. Not always like maybe it is time between them signing off and me signing on. Like for example say 4 hours between us, everything broken, 1 hour between us stuff still working. I can usually tell.if they have been on recently because I am full power. Which if they haven't been on I have no power. 

However the part where I disagree. Sometimes everything goes off and quits working if I leave and go to another dimension for a while.not even that I guess. There have been times where I left the base off to raid pirate ships and stuff for a break. I get back to base and I am out of power and nothing working. However this also only occurs if I am only one on. It seems if the server is hopping, all 6 of us on and going and doing things. I leave my base for long periods of time or go to other dimensions things stay working. We do things now like just stay logged in, get in a mine cart doing circles to stay active.

**Comment by DietmarKrause** [comment]
perhaps related: I have found on Single player world (All the mods 1, latest upddate, single player) , that the refined storage importer / exporter do not work any more if i get to the nether and return. All chuncs in my base are chunc loaded via ftbu chuncloading. simple restart of the world helps. I will try a different method of chuncloading and i will have to see, if the chunc borders are crossed somehow.

**Comment by DietmarKrause** [comment]
In my case (SP World, modded 1.10.2) problem is solved with chicken chunks chunkloader instead of ftb utils. Tested in 2 Worlds (enderio conduits problem and refinded storage importer and exporter work fine after that change)

**Comment by bound4doom** [comment]
@DietmarKrause How did you get chicken chunks to work? Anything special on config changes? We tried getting it to load but it would crash the world on the server. So we abandoned it. This issue with chunk loading is driving us all nuts. 

@LatvianModder So we have a lot of testing results. They took a lot longer than expected because well now we have a whole server crash. So Yes, restarting the server, some chunks load, not all of them, just some of them. Our latest crash is a Refined Storage crash so bad it will not let the server actually start. However, what is causing the crash is, from the logs we can tell refined storage is trying to auto craft something with fluid that is not chunk loaded. Even though we have everything chunk loaded where it would be pulling fluid from. We haven't quite figure out what yet as it is time consuming. We have to completely download the whole world, go in with MCEdit break the refined storage controller then upload the world restart the server, we can then log in. However, we need to create a new controller then get everything running again. So we kind of like to play some. 

We do play and everything seems to stay working, we stopped auto rebooting. However if you do leave the area like go off to the nether, the end, another dimension and so on.  Somethings will break still and it seems some chunks unload. However no world crash. Once we restart the server, though, the world will crash, refined storage again trying to autocraft with liquid not chunk loaded. So we have stopped all autocrafting using liquid for now . We will try reintroducing them slowly over time the next few days. To see if we can find what it is. Like I said we still kind of want to play and when we crash the world it is a huge pain.

**Comment by bound4doom** [comment]
More of an update today from the weekend. I would say you are correct in your initial assessment chunks are not being loaded after server restart. However, I did experience things breaking at the chunk borders again when leaving the area. Things have been good for a while, I haven't had a lot of issues, more simple ones break a pipe and reconnect it things start working again after I sign on. However, I have been doing a lot of building and constructing on my own base and haven't needed to leave it. 

I needed some stuff from the Nether, now I have been in and out of the nether several times lately just to kill some mobs and get back out. No issues. However I needed to mine there, so I set up a builder, claimed and chunkloaded the chunks it would mine and then kicked it off and let it go. I wasn't in there for very long. Less time than a normal hunting trip. I got back and things were broken again. So maybe the issue might also have to do with having chunks loaded in multiple dimensions? Or unconnected chunks. So like claim a few chunks here, claim a few chunks there.

**Comment by LatvianModder** [comment]
From what I have tested it seems like:

+ It doesn't re-load the chunks after server restart, until you log in
+ Only loads chunks after you log off, if you are an OP

I think the problem is with permissions / rank configs, and I'm looking into that. Best way to rest chunkloading btw is to set up a Counter from RFTools with Redstone Clock from XU2, set max to 999 and see how many actual seconds have passed since you logged off

**Comment by LatvianModder** [comment]
I think I might just remove "unload after x time" because that messes with chunkloading too much. For now there will be 2 options - either loaded all the time or unloads after player logs off

**Comment by joshuagore** [comment]
Do you think the unload after x time is what is causing the unload to occur when I walk away from the chunks without having logged off?

**Comment by LatvianModder** [comment]
Maybe not that specifically, but there's just too many variables that the chunk manager has to check. Im going to try to simplify it, maybe then I can also fix that other problem

**Comment by joshuagore** [comment]
Mmmmkay, if you need help hunting anything else down, let me know and I can try things out.

**Comment by LatvianModder** [comment]
Try now with v3.5.0

**Comment by bound4doom** [comment]
I can't test on our server until a new release of Direworlf 20 mod pack as they will only install released versions. I can try this weekend on single client and see.

**Comment by jarquafelmu** [comment]
I can confirm that this issue still persists in FTBUtils-1.10.2-3.5.1, every so often (about 15 minutes or so), my friend's enderio conduits stop working and he has to break a conduit to make them work again. We are using only the FTBUtilies chunk loader because the other ones are disabled.

We are running FTB Beyond 1.4.2. If you need any more information please ask.

**Comment by ChaoticTabris** [comment]
I just removed FTBUtils and the problem persists. FTBUtils might not be the culprit afterall

**Comment by HenryLoenwind** [comment]
@ChaoticTabris FTBUtils is not the sole source, it is just very good in triggering it. The latest Ender IO build (183) should not be that receptive to it anymore.

**Comment by LatvianModder** [comment]
It's not even good at triggering it. People usually don't realize that mostly chunkloader blocks load a whole bunch of chunks around them, like 25 (5x5) or at least 9 (3x3), but FTBUtilities only loads the chunks you select to be loaded. So if you load a chunk and chunk next to it isn't loaded, most likely you will encounter problems

**Comment by Anarcraft** [comment]
I'm on FTB Beyond 1.5.3 (with Sponge, not sure which version but I could ask the admin) having an issue with chunks unloading after a few minutes (1-5 minutes) after leaving the area but still being in the overworld.  I tried your suggestion of after a server restart log on, then log off, wait some time, then log back on and I got some very interesting results indicating the chunks loading but still having border issues.

My base is a 3x3 chunk area.  On one of the corner chunks is my ender chest with a conduit that goes to an ME interface and the power systems are also in this chunk.  The AE controllers and storage are all in the center chunk of the 3x3.  Upon doing the log on/off/on test after server restart I found that finally the chunks are staying loaded (items from the ender chest were being pulled out and my infinite range wireless AE was working).  I hypothesize that all 9 chunks are actually loaded but not allowing "cross communication" so the Ender IO conduits wouldn't link up across chunk borders.  After a few minutes the wireless AE stops working (my big controller runs out of internal power) yet the ender chest will forever put items into the ME interface (which is now offline but still had some internal buffer room).
